### PR TITLE
Improve elm linter

### PR DESCRIPTION
**Pull request by jwoudenberg**
Currently some elm compiler errors are not returned in json format. An example of this is the case of importing a module that does not exist. The current behavior of the elm-make ale linter is to simply not show an error in this case requiring one to manually run elm-make to figure out what's broken.

This PR adds a fallback. In case of a non-parseable error it takes the error-message as is and attaches it on the first line.

### CODE REVIEW

File: ale_linters/elm/make.vim [hunk 1]
```diff
@@ -29,9 +30,22 @@ function! ale_linters#elm#make#Handle(buffer, lines) abort
...
                 endif
             endfor
+        elseif l:line !=# 'Successfully generated /dev/null'
+            call add(l:unparsed_lines, l:line)
         endif
     endfor
 
+    if len(l:unparsed_lines) > 0
+        call add(l:output, {
+        \    'bufnr': a:buffer,
```
Comment (w0rp) [inline ale_linters/elm/make.vim]: This key doesn't need to be added.

File: ale_linters/elm/make.vim [hunk 2]
```diff
@@ -29,9 +30,22 @@ function! ale_linters#elm#make#Handle(buffer, lines) abort
...
+        \    'lnum': 0,
+        \    'col': 0,
```
Comment (w0rp) [inline ale_linters/elm/make.vim]: You don't need to set this key if there's no column.

File: ale_linters/elm/make.vim [hunk 3]
```diff
@@ -29,9 +30,22 @@ function! ale_linters#elm#make#Handle(buffer, lines) abort
...
             endfor
+        elseif l:line !=# 'Successfully generated /dev/null'
+            call add(l:unparsed_lines, l:line)
         endif
     endfor
 
+    if len(l:unparsed_lines) > 0
+        call add(l:output, {
+        \    'bufnr': a:buffer,
+        \    'lnum': 0,
```
Comment (w0rp) [inline ale_linters/elm/make.vim]: You should set the line number to `1` if the error is at the first line.

**Comment by jwoudenberg** [comment]
Thanks for the review @w0rp. I've got a lot going on this week, will return to this next!

**Comment by jwoudenberg** [comment]
Thank you for the review!

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**